### PR TITLE
Merge upstream 9.8

### PR DESCRIPTION
**Pull request by tgauth**
# PR Summary
- pull in changes from upstream 9.8

## PR Context
Windows specific changes:
- add `contrib\win32\openssh\sshd-session.vcxproj` 
- add `contrib\win32\win32compat\wmain_sshd-session.c`
- tweak `sshd` to still support `debug_flag` and `inetd_flag` with `sshd-session`
- set default `_PATH_SSHD_SESSION` to sshd-session.exe and de-relativize path in `servconf.c`
- some `log.c` removals surface some compilation errors (false positives) related to uninitialized vars in `ssh-sk-helper.c` and `ssh-pkcs11-client.c`, but add initializations with #ifdefs 
- define signals added upstream as unsupported in `contrib/win32/win32compat/inc/signal.h`

### CODE REVIEW

**Comment by gajanan-choudhary** [comment]
Do we know if this PR should fix CVEs listed at [openssh.com/security.html](https://www.openssh.com/security.html) or [cvedetail.com](https://www.cvedetails.com/vulnerability-list/vendor_id-97/product_id-585/version_id-1774819/Openbsd-Openssh-9.6.html)? We need those fixed to be allowed to use to OpenSSH in our Windows infrastructure.

**Comment by tgauth** [comment]
> Do we know if this PR should fix CVEs listed at [openssh.com/security.html](https://www.openssh.com/security.html) or [cvedetail.com](https://www.cvedetails.com/vulnerability-list/vendor_id-97/product_id-585/version_id-1774819/Openbsd-Openssh-9.6.html)? We need those fixed to be allowed to use to OpenSSH in our Windows infrastructure.

Please see https://github.com/PowerShell/Announcements/issues/63 regarding CVE-2024-6387.

File: servconf.c [hunk 1]
```diff
@@ -1871,38 +1871,9 @@ process_server_config_line_depth(ServerOptions *options, char *line,
...
-			else
-			{
-				debug3_f("'%s' not recognized as SID", arg);
-			}
-#endif // WINDOWS
-			if (*arg == '\0' ||
-				match_user(NULL, NULL, NULL, arg) == -1)
-					fatal("%s line %d: empty %s pattern",
-					filename, linenum, keyword);
+			if (*arg == '\0')
```
Comment (vthiebaut10) [inline servconf.c]: What is this change for?

Comment (tgauth) [reply servconf.c]: this fixes the duplicate issue from merging - the `#ifdef WINDOWS` code is still defined below

**Review by anmenaga — APPROVED** [review]

**Pull request merged**